halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/NestableCommand.java

```diff
@@ -18,44 +18,77 @@
 
 import com.beust.jcommander.JCommander;
 import com.beust.jcommander.Parameter;
+import com.netflix.spinnaker.halyard.cli.ui.v1.Ui;
+import com.netflix.spinnaker.halyard.config.errors.v1.HalconfigFixableIssue;
+import lombok.AccessLevel;
+import lombok.Getter;
 import lombok.Setter;
+import retrofit.RetrofitError;
 
+import java.net.ConnectException;
 import java.util.Map;
 
 abstract class NestableCommand {
   @Setter
-  protected JCommander commander;
-  private GlobalOptions globalOptions;
+  @Getter(AccessLevel.PROTECTED)
+  private JCommander commander;
+
+  @Getter(AccessLevel.PROTECTED)
+  private Ui ui;
 
   @Parameter(names = { "-h", "--help" }, help = true)
   private boolean help;
 
-  public NestableCommand(GlobalOptions globalOptions) {
-    this.globalOptions = globalOptions;
+  NestableCommand() {
+    boolean color = GlobalOptions.getGlobalOptions().isColor();
+    ui = new Ui(color);
   }
 
   /**
-   * This recusively walks the chain of subcommands, until it finds the last in the chain, and runs executeThis.
+   * This recursively walks the chain of subcommands, until it finds the last in the chain, and runs executeThis.
    *
    * @see NestableCommand#executeThis()
    */
   public void execute() {
     String subCommand = commander.getParsedCommand();
     if (subCommand == null) {
-      executeThis();
+      safeExecuteThis();
     } else {
       getSubcommands().get(subCommand).execute();
     }
   }
 
-  abstract protected void executeThis();
+  /**
+   * Used to consistently format exceptions thrown by connecting to the halyard daemon.
+   */
+  private void safeExecuteThis() {
+    try {
+      executeThis();
+    } catch (RetrofitError e) {
+      if (e.getCause() instanceof ConnectException) {
+        ui.failure(e.getCause().getMessage());
+        ui.remediation("Is your daemon running?");
+      } else {
+        HalconfigFixableIssue issue = (HalconfigFixableIssue) e.getBodyAs(HalconfigFixableIssue.class);
+        for (String error : issue.getErrors()) {
+          ui.failure(error);
+        }
+
+        for (String warning : issue.getWarnings()) {
```

Comment:
FWIW, I like errors after warnings (increasing severity). That way when you finish scrolling and just look at the bottom, you see the worst case. I guess it could also keep the suspense reading linearly.